src/nti/externalization/externalization.py

```diff
@@ -285,14 +285,20 @@ def _to_external_object_state(obj, state, top_level=False, decorate=True, useCac
 						 type(obj), obj.__class__)
 		return state.catch_component_action(obj, t)
 
-def toExternalObject(obj, coerceNone=False, name=_NotGiven, registry=component,
-					  catch_components=(), catch_component_action=None,
-					  default_non_externalizable_replacer=DefaultNonExternalizableReplacer,
-					  request=_NotGiven,
-					  decorate=True,
-					  useCache=True,
-					  decorate_callback=None):
-	""" Translates the object into a form suitable for
+def toExternalObject(obj, 
+					 coerceNone=False, 
+					 name=_NotGiven, 
+					 registry=component,
+					 catch_components=(), 
+					 catch_component_action=None,
+					 default_non_externalizable_replacer=DefaultNonExternalizableReplacer,
+					 request=_NotGiven,
+					 decorate=True,
+					 useCache=True,
+					 decorate_callback=None,
+					 **kwargs):
```

Comment:
Why did we make this start taking arbitrary keyword arguments that it doesn't do anything with?

Calling a function that takes `**kwargs` is slower than one that doesn't (and that's before we deal with the updated bookkeeping):
```pycon
In [1]: def no_kw(): pass

In [2]: def with_kw(a=1): pass

In [3]: def with_kw_arbitrary(a=1, **kwargs): pass

In [4]: %timeit no_kw()
The slowest run took 7.16 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 166 ns per loop

In [5]: %timeit with_kw()
The slowest run took 11.67 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 184 ns per loop

In [6]: %timeit with_kw_arbitrary()
The slowest run took 9.66 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 3: 197 ns per loop
```

Reply:
I don't have anything useful to add in terms of why that is there or what is being done with it, but anywhere in this code we can shave any time would be ideal.  Of the profiles I've looked at most recently externalization has been a warm spot (not typically the hot spot) in many of them especially when there are lots of nested objects. Every little bit helps I assume.

Reply:
Yeah, it's always been pretty warm. After we get coverage up to 100% where we can feel comfortable making more substantial changes, there might be some ways to improve things...we've learned quite a lot since all this code was originally written and I bet we can do better (I'm seeing things in here I wrote that are making me cringe).

Or we could just use PyPy 😄 It's over three-times as fast in this microbenchmark: 
```pycon
In [2]: %timeit no_kw()
The slowest run took 133.19 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 7: 53.7 ns per loop

In [3]: %timeit with_kw()
The slowest run took 52.05 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 7: 55 ns per loop

In [4]: %timeit with_kw_arbitrary()
The slowest run took 66.04 times longer than the fastest. This could mean that an intermediate result is being cached.
10000000 loops, best of 7: 61.4 ns per loop
```

Reply:
This is a good point. Performance concerns keep cropping up lately. What's keeping us from running on PyPY at this point?